src/main/java/io/seqera/migtool/MigTool.java

```diff
@@ -350,7 +360,9 @@ protected void apply() {
 
     protected void checkRank(MigRecord entry) {
         try(Connection conn=getConnection(); Statement stm = conn.createStatement()) {
-            ResultSet rs = stm.executeQuery("select max(`rank`) from "+MIGTOOL_TABLE);
+            String rank = dialect.isPostgres() || dialect.isTestContainersPostgres()  ? "rank" : "`rank`";
```

Comment:
As an alternative to defining `column` and `\`column`\`, we could add a method such as `transformColumnsToPostgres` which removes the backticks. Ideally, the method will have a comment explaining that backticks are not supported in PostgreSQL.

Reply:
I agree this should be improved. Also, it could be considered a kind of "driver" pattern.

Reply:
Actually I think that a strategy pattern can be applied easily here, however I went for the fast fix. I will improve the solution then.